library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/DispatcherLocationProvider.java

```diff
@@ -192,9 +214,14 @@ void continueWithDefaultProviders() {
                 getListener().onLocationFailed(FailType.GOOGLE_PLAY_SERVICES_NOT_AVAILABLE);
             }
         } else {
-            LogUtils.logI("Attempting to get location from default providers...");
-            setLocationProvider(getSourceProvider().createDefaultLocationProvider());
-            activeProvider.get();
+            // check if activeProvider is already DefaultLocationProvider
+            if (activeProvider instanceof DefaultLocationProvider) {
```

Comment:
Any reason for adding this check? Also, what is the expected behaviour if it is already default provider?